azurerm/internal/services/apimanagement/api_management_identity_provider_aadb2c_resource.go

```diff
@@ -0,0 +1,228 @@
+		},
+	}
+}
+
+func resourceArmApiManagementIdentityProviderAADB2CCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.IdentityProviderClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("api_management_name").(string)
+
+	clientID := d.Get("client_id").(string)
+	clientSecret := d.Get("client_secret").(string)
+
+	allowedTenant := d.Get("allowed_tenant").(string)
+	signinTenant := d.Get("signin_tenant").(string)
+	authority := d.Get("authority").(string)
+	signupPolicy := d.Get("signup_policy").(string)
+
+	signinPolicy := d.Get("signin_policy").(string)
+	profileEditingPolicy := d.Get("profile_editing_policy").(string)
+	passwordResetPolicy := d.Get("password_reset_policy").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, serviceName, apimanagement.AadB2C)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Identity Provider %q (API Management Service %q / Resource Group %q): %s", apimanagement.AadB2C, serviceName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_api_management_identity_provider_aadb2c", *existing.ID)
+		}
+	}
+
+	parameters := apimanagement.IdentityProviderCreateContract{
+		IdentityProviderCreateContractProperties: &apimanagement.IdentityProviderCreateContractProperties{
+			ClientID:                 utils.String(clientID),
+			ClientSecret:             utils.String(clientSecret),
+			Type:                     apimanagement.AadB2C,
+			AllowedTenants:           utils.ExpandStringSlice([]interface{}{allowedTenant}),
+			SigninTenant:             utils.String(signinTenant),
+			Authority:                utils.String(authority),
+			SignupPolicyName:         utils.String(signupPolicy),
+			SigninPolicyName:         utils.String(signinPolicy),
+			ProfileEditingPolicyName: utils.String(profileEditingPolicy),
+			PasswordResetPolicyName:  utils.String(passwordResetPolicy),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, serviceName, apimanagement.AadB2C, parameters, ""); err != nil {
+		return fmt.Errorf("creating or updating Identity Provider %q (Resource Group %q / API Management Service %q): %+v", apimanagement.AadB2C, resourceGroup, serviceName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, serviceName, apimanagement.AadB2C)
+	if err != nil {
+		return fmt.Errorf("retrieving Identity Provider %q (Resource Group %q / API Management Service %q): %+v", apimanagement.AadB2C, resourceGroup, serviceName, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read ID for Identity Provider %q (Resource Group %q / API Management Service %q)", apimanagement.AadB2C, resourceGroup, serviceName)
+	}
+	d.SetId(*resp.ID)
```

Comment:
rather than using the API ID here, can we use the ID from the Parser/Formatter here? this'd also mean we don't need this additional Get request here